corehq/motech/tests/test_value_source.py

```diff
@@ -634,3 +654,32 @@ def test_set_external_data(self):
 def test_doctests():
     results = doctest.testmod(corehq.motech.value_source, optionflags=doctest.ELLIPSIS)
     assert results.failed == 0
+
+
+def _create_case(case, utc_datetime) -> CommCareCaseSQL:
```

Comment:
It's a shame it takes so much overhead just to create a case for a test (although I don't know a better way). I wonder if it would make sense to move it into a test helper module so other tests can use it too? Might be able to eliminate some uses of `submit_form_locally` (which is probably way slower than this) in the process...

Reply:
Another option would be to use `CaseBlock` and `submit_case_block` (which uses `submit_form_locally`, and is probably slower).

https://github.com/dimagi/commcare-hq/blob/7f5a05086678f63928304ec0f9062fdeedd2ed3c/corehq/apps/hqcase/tests/test_serialization.py#L36-L62

You were probably trying to avoid that though (which seems fine).